File: mobile/ios/Sources/YDKHubClient/YDKError.swift
import Foundation

/// yuleDKCS SDK 错误类型
public enum YDKError: Error, LocalizedError {
    case notInitialized
    case notAuthenticated
    case hubError(String, String)   // (errorCode, errorMsg)
    case httpError(Int)             // HTTP 状态码错误 (4xx/5xx)
    case networkError(Error)
    case timeout
    case decodingFailed(String)     // JSON 解析失败
    case internal_(String)

    public var errorDescription: String? {
        switch self {
        case .notInitialized:        return "SDK 未初始化"
        case .notAuthenticated:      return "未登录，请先调用 setToken()"
        case .hubError(let code, let msg): return "[\(code)] \(msg)"
        case .httpError(let code):   return "HTTP 错误: \(code)"
        case .networkError(let e):   return "网络错误: \(e.localizedDescription)"
        case .timeout:               return "请求超时"
        case .decodingFailed(let d): return "JSON 解析失败: \(d)"
        case .internal_(let msg):    return "SDK 内部错误: \(msg)"
        }
    }
}

// MARK: - Hub REST Gateway 错误响应格式
public struct HubErrorResponse: Decodable {
    public let error: String?
    public let message: String?
    public let code: String?
}

File: mobile/android/sdk/src/main/kotlin/com/yuledkcs/sdk/hub/YDKError.kt
package com.yuledkcs.sdk.hub

/**
 * yuleDKCS SDK 错误类型
 */
sealed class YDKError : Exception() {
    object NotInitialized : YDKError()
    object NotAuthenticated : YDKError()
    data class HubError(val code: String, val msg: String) : YDKError()
    data class HttpError(val statusCode: Int) : YDKError()
    data class NetworkError(val cause: Throwable) : YDKError()
    object Timeout : YDKError()
    data class DecodingFailed(val detail: String) : YDKError()
    data class Internal(val msg: String) : YDKError()

    override val message: String get() = when (this) {
        NotInitialized -> "SDK 未初始化"
        NotAuthenticated -> "未登录，请先调用 setToken()"
        is HubError -> "[$code] $msg"
        is HttpError -> "HTTP 错误: $statusCode"
        is NetworkError -> "网络错误: ${cause.message}"
        Timeout -> "请求超时"
        is DecodingFailed -> "JSON 解析失败: $detail"
        is Internal -> "SDK 内部错误: $msg"
    }
}

/** Hub REST Gateway 错误响应格式 */
data class HubErrorResponse(
    val error: String? = null,
    val message: String? = null,
    val code: String? = null
)
